source/fundamentals/auth.txt

```diff
@@ -10,22 +10,254 @@ Authentication Mechanisms
    :depth: 2
    :class: singlecol
 
-In this guide, we show you how to authenticate with MongoDB using each
-**authentication mechanism** available in the MongoDB Community Edition.
-Authentication mechanisms are processes by which the driver and server
-confirm identity and establish trust to ensure security.
-
-The mechanisms that you can use with the latest version of MongoDB Community
-Edition are as follows:
-
-* Default
-* SCRAM-SHA-256
-* SCRAM-SHA-1
-* MONGODB-CR
-* MONGODB-AWS
-* X.509
-
-To authenticate using ``Kerberos`` or ``LDAP``, see the
-:doc:`Enterprise Authentication Mechanisms guide </fundamentals/enterprise-auth>`.
-For more information on establishing a connection to your MongoDB cluster,
-read our :doc:`Connection Guide </fundamentals/connection>`.
+Overview
+--------
+
+In this guide, you can learn how to use each **authentication
+mechanism** available in the MongoDB Community Edition. MongoDB uses
+authentication mechanisms to confirm an identity and establish trust to
+ensure security in the driver and server before connecting.
+
+If you want to establish a connection to your MongoDB cluster without an
+authentication mechanism, see our :doc:`Connection Guide </fundamentals/connection>`.
+
+Supported Mechanisms
+--------------------
+
+The Go Driver establishes a connection with an authentication mechanism
+through a `Client <https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.1/mongo#Client>`__
+type. The ``Client`` type specifies the mechanism and credentials to use
+as connection options in a `Credential <https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.1/mongo/options#Credential>`__
+type . To configure these options, pass a ``Credential`` type to a  `ClientOptions
+<https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.1/mongo/options#ClientOptions>`__
+type. 
+
+The following sections demonstrate this process with the five
+mechanisms the MongoDB Community Edition supports. 
+
+Example Conventions
+~~~~~~~~~~~~~~~~~~~
+
+Each authentication mechanism contains the following placeholders: 
+
+* ``username`` - Your MongoDB username
+* ``password`` - Your MongoDB user's password
+* ``hostname`` - Your MongoDB servers network address, accessible by
+  your client 
+* ``port`` - Your MongoDB servers port number
+* ``authenticationDb`` - Your MongoDB database that contains the user's
+  authentication data. If you omit this option, the driver uses the
+  default value ``admin``.
+
+.. _default-auth-mechanism:
+
+Default
+~~~~~~~
+
+The default mechanism uses one of the following authentication
+mechanisms depending on what MongoDB versions your server supports:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :class: compatibility-large
+
+   * - Mechanism
+     - Versions
+
+   * - ``SCRAM-SHA-256``
+     - MongoDB 4.0 and later   
+   
+   * - ``SCRAM-SHA-1``
+     - MongoDB 3.0, 3.2, 3.4, and 3.6
+
+   * - ``MONGODB-CR``
+     - MongoDB 3.6 and earlier
```

Comment:
I believe this should be 2.6 and earlier.

Reply:
hm, I found the mention in the [server manual release notes ](https://docs.mongodb.com/manual/release-notes/3.6/#security) and it says it was depreciated as of 3.6 -- maybe the edit is "earlier than MongoDB 3.6"?

Reply:
Based on [`chooseAuthMechanism`](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/auth/default.go#L71-L73) it looks like we pick `SCRAM-SHA-256` if it's supported, falling back to `SCRAM-SHA-1` if not, and finally to `MONGODB-CR` if neither is supported.

Reply:
The link you provided contains `/x`, which I was told is experimental and not for public use.

Reply:
As discussed offline: most of our authentication logic is "unstable" API and is therefore under `/x`. I wouldn't point to code under `/x` in documentation, but I was just linking to that code here to explain how the Go driver actually picks the "default" auth mechanism.